ui/app/models/clients/config.js

```diff
@@ -4,32 +4,47 @@
  */
 
 import Model, { attr } from '@ember-data/model';
-import { computed } from '@ember/object';
-import attachCapabilities from 'vault/lib/attach-capabilities';
-import { expandAttributeMeta } from 'vault/utils/field-to-attrs';
-import { apiPath } from 'vault/macros/lazy-capabilities';
-
-const M = Model.extend({
-  queriesAvailable: attr('boolean'), // true only if historical data exists, will be false if there is only current month data
-  retentionMonths: attr('number', {
+import lazyCapabilities, { apiPath } from 'vault/macros/lazy-capabilities';
+import { withFormFields } from 'vault/decorators/model-form-fields';
+import { withModelValidations } from 'vault/decorators/model-validations';
+
+const validations = {
+  retentionMonths: [
+    {
+      validator: (model) => parseInt(model.retentionMonths) >= (model.reportingEnabled ? 24 : 0),
+      message: (model) => {
+        if (model.reportingEnabled) {
+          return 'Retention period must be a minimum of 24 months.';
+        }
+        return 'Retention period must be greater than or equal to 0.';
```

Comment:
I discussed a nit I had with @zofskeez out-of-band: 

These errors will be returned by the backend. If we include them statically here, we'll have to make sure they're updated if the contraints are ever updated on the backend. I wonder if there's a way to use the backend errors instead?

Reply:
To align with the designs I wonder if we could leverage the current model validations pattern/object shape and maybe create a util that will convert API errors into the validations format that can be passed in to `FormField`?

Reply:
I think unless the errors from the API can be returned as an object with the key names mapped to the properties that have errors we won't actually know which form field to display the message on though.

Reply:
One idea to still leverage the backend errors is to instead `warn` here, so submission is still possible.  This deviates from the design patterns a bit, but just throwing it out there

Reply:
Ya that could work but I'm not sure if it's preferable. If the backend value were to change then the form validation might warn that the minimum is 24 and then the API error would say something different like 36 in that message which may be confusing.

Reply:
Yeah, exactly. The subtext for the input also contains the minimum value, right? So either way we have to update the UI if/when the backend changes. Maybe a note in the backend code to update the corresponding UI code (or let the UI team know) if the value is ever changed, @mpalmi ?

Reply:
Yeah, that was one option I considered. Another would be to throw the minimum retention months in the config endpoint. I'm okay with either approach. I'll throw a comment in my open PR for now and we can come back to this if we ever need to!